app/models/job-link.ts

```diff
@@ -0,0 +1,162 @@
+import _ from 'lodash';
+import { Transaction } from 'util/db';
+import { removeEmptyProperties } from 'util/object';
+import Record from './record';
+
+interface BaseJobLink {
+  id?: number;
+  jobID?: string;
+  href: string;
+  type?: string;
+  title?: string;
+  rel?: string;
+  createdAt?: Date;
+  updatedAt?: Date;
+}
+
+interface JobLinkRecord extends BaseJobLink {
+  bbox?: string;
+  temporalStart?: Date;
+  temporalEnd?: Date;
+}
+
+/**
+ * For the constructor we can take either a JobLink or a JobLinkRecord.
+ * When created from the database it will be a JobLinkRecord, but when
+ * constructed outside of the database it will likely be a JobLink
+ * which has a different representation for the bbox and temporal.
+ * Specifying JobLink | JobLinkRecord as the type causes many Typescript
+ * errors so we use this interface to avoid that.
+ */
+
+export interface JobLinkOrRecord extends BaseJobLink {
+  bbox?: string | number[];
+  temporalStart?: Date;
+  temporalEnd?: Date;
+  temporal?: {
+    start?: Date;
+    end?: Date;
+  };
+}
+
+export interface SerializedJobLink {
+  href: string;
+  type?: string;
+  title?: string;
+  rel?: string;
+  temporal?: {
+    start?: string;
+    end?: string;
+  };
+  bbox?: number[];
+}
+
+const serializedLinkFields = ['href', 'type', 'title', 'rel', 'bbox'];
+
+const jobLinkCommonFields = ['id', 'jobID', 'createdAt', 'updatedAt', 'href', 'type', 'title', 'rel'];
+
+/**
+ *
+ * Wrapper object for persisted job links
+ *
+ */
+export default class JobLink extends Record {
+  static table = 'job_links';
+
+  jobID: string;
+
+  href: string;
+
+  type?: string;
+
+  title?: string;
+
+  rel?: string;
+
+  temporal?: {
+    start?: Date;
+    end?: Date;
+  };
+
+  bbox?: number[];
+
+  /**
+   * Creates a Job link from the links in a job.
+   *
+   * @param fields - Object containing fields to set on the record
+   */
+  constructor(fields: JobLinkOrRecord) {
```

Comment:
I think it might have been better to use constructor overloading and then in the implementation use `typeof` to check the type, rather than introducing the `JobLinkOrRecord` class.

Reply:
That interface was needed in order to compile without type errors.

Can you give me an example of what you're suggesting though (that's what I wanted to do to, but I couldn't figure out how to make it work because Typescript doesn't actually let you define multiple constructors that take different types).

Reply:
It does let you _declare_ multiple constructors. You can only have one implementation, but that can be generic and can do different things depending on the type of the input:
```typescript
constructor(fields: JobLink); 
constructor(fields: Record);
constructor(fields: JobLink | Record) {
    super(fields);
    if (fields instanceof JobLink) {
       // init with JobLink
    } else {
      // init with Record
    }
```

Reply:
It has the same issues that caused me to introduce the interface:

```
Property 'temporalStart' does not exist on type 'JobLinkRecord | JobLink'.
  Property 'temporalStart' does not exist on type 'JobLink'.
```